File: multiple_inheritance.py
# Multiple Inheritance occurs when a class inherits from more than one base class.

class Mammal:
    def feed_milk(self):
        print("Feeding milk")


class Bird:
    def lay_eggs(self):
        print("Laying eggs")


class Bat(Mammal, Bird):
    pass


bat = Bat()
bat.feed_milk()
bat.lay_eggs()

File: multiple_inheritance.cpp
// Multiple Inheritance occurs when a class inherits from more than one base class.

#include <bits/stdc++.h>
using namespace std;

class Mammal
{
public:
    void feedMilk()
    {
        cout << "Feeding milk" << endl;
    }
};

class Bird
{
public:
    void layEggs()
    {
        cout << "Laying eggs" << endl;
    }
};

class Bat : public Mammal, public Bird
{
};

int main()
{
    Bat obj;
    obj.feedMilk();
    obj.layEggs();
    return 0;
}
